python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py

```diff
@@ -240,7 +244,16 @@ def update_replicas(self, replicas: List[ReplicaWrapper]):
         new_replica_id_set = set()
         new_colocated_replica_ids = defaultdict(set)
         new_multiplexed_model_id_to_replica_ids = defaultdict(set)
+
         for r in replicas:
+            # If on the proxy, replica needs to call back into the proxy with
+            # `receive_asgi_messages` which can be blocked when GCS is down.
+            # To prevent that from happening, push proxy handle eagerly
+            if self._handle_source == DeploymentHandleSource.PROXY:
+                r._actor_handle.push_proxy_handle.remote(
```

Comment:
let's add a method to the interface, shouldn't be accessing the `_actor_handle` private attribute

Reply:
that way it can be tested as well